react/src/components/atoms/forms/InputTextFuzzy/index.js

```diff
@@ -0,0 +1,149 @@
+                  return parts.filter(part => {
+                    return part.text.length >= 1;
+                  }).map((part, index) => {
+                    const className = part.highlight ? 'highlight' : null;
+                    const key = `${match.key}.suggestion_${index}`;
+                    return(
+                      <span className={className} key={key}>{part.text}</span>
+                    );
+                  });
+                }
+              })
+            }
+          </span>
+        </span>
+    );
+  };
+  renderItemsContainer = ({ children, containerProps }) => {
+    return(<div className="ma__input-fuzzy" {...containerProps}>{children}</div>)
+  };
+  render() {
+    const autoProps = {
+      items: this.state.suggestions,
+      renderItemsContainer: this.renderItemsContainer,
+      renderItem: this.renderItem,
+      renderItemData: { value: this.state.value },
+      inputProps: {
+        placeholder: this.props.placeholder,
+        onChange: this.handleChange,
+        value: this.state.value,
+        disabled: this.props.disabled
+      },
+      id: this.props.id
+    };
+    autoProps.itemProps = (props) => {
+      const { itemIndex } = props;
+      const suggestion = this.state.suggestions[itemIndex];
+      return {
+        'data-item-index': itemIndex,
+        onMouseDown: event => {
+          this.setState({
+            value: suggestion.item.text,
+            suggestions: []
+          });
+          if (typeof this.props.onSuggestionClick === 'function') {
+            // Suggestion is an object that can contain info on score, matches, etc.
+            this.props.onSuggestionClick({event, suggestion});
+          }
+        }
+      }
+    };
+    const inputTextTypeAheadClasses = classNames({
+      'ma__input-typeahead': true,
+      'ma__input-typeahead--disabled': this.props.disabled,
+      'ma__input-typeahead--boxed': this.props.boxed,
+    });
+    return(
+      <React.Fragment>
+        {this.props.label && (<label htmlFor={this.props.id} className="ma__label">{this.props.label}</label>)}
+        <div className={inputTextTypeAheadClasses}>
+          <Autowhatever {...autoProps} />
+        </div>
+      </React.Fragment>
+    );
+  }
+}
+
+InputTextFuzzy.propTypes = {
+  /** The id of the typeahead element. */
+  id: PropTypes.string.isRequired,
+  /** The label text above the input text box. */
+  label: PropTypes.string,
+  /** The placeholder text to appear in the input text box. */
+  placeholder: PropTypes.string,
+  /** Style the input with a box outline. */
+  boxed: PropTypes.bool,
+  /** The keys within options to use with search. */
+  keys: PropTypes.arrayOf(PropTypes.string).isRequired,
+  /** An array of objects representing all searchable values. */
+  options: PropTypes.arrayOf(PropTypes.object).isRequired,
+  /** Any Fusejs options to override the default options set in this component.*/
+  fuseOptions: PropTypes.object,
```

Comment:
@smurrayatwork --> can you add an md file to this, specifically it would be nice to link to the fuse packages options in there so future users know where to look.